codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -396,4 +405,228 @@ public TextHeaders clear() {
+                charSequenceEscaper = new CsvValueEscaper<CharSequence>() {
+                    @Override
+                    public CharSequence escape(CharSequence value) {
+                        return StringUtil.escapeCsv(value);
+                    }
+                };
+            }
+            return charSequenceEscaper;
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence value) {
+            return addEscapedValue(name, StringUtil.escapeCsv(value));
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence... values) {
+            return addEscapedValue(name, commaSeparate(charSequenceEscaper(), values));
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, Iterable<? extends CharSequence> values) {
+            return addEscapedValue(name, commaSeparate(charSequenceEscaper(), values));
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Iterable<?> values) {
+            return addEscapedValue(name, commaSeparate(objectEscaper(), values));
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Object... values) {
+            return addEscapedValue(name, commaSeparate(objectEscaper(), values));
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, CharSequence... values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(charSequenceEscaper(), values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(charSequenceEscaper(), values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(objectEscaper(), values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(objectEscaper(), values));
+            return DefaultTextHeaders.this;
+        }
+
+        private TextHeaders addEscapedValue(CharSequence name, CharSequence escapedValue) {
+            CharSequence currentValue = DefaultTextHeaders.super.get(name);
+            if (currentValue == null) {
+                DefaultTextHeaders.super.add(name, escapedValue);
+            } else {
+                DefaultTextHeaders.super.set(name, commaSeparateEscapedValues(currentValue, escapedValue));
+            }
+            return DefaultTextHeaders.this;
+        }
+
+        private <T> CharSequence commaSeparate(CsvValueEscaper<T> escaper, T... values) {
+            StringBuilder sb = new StringBuilder(values.length * DEFAULT_VALUE_SIZE);
```

Comment:
Why create new StringBuilders instead of using pooled `InternalThreadLocalMap#stringBuilder()`?